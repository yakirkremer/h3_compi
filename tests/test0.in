int z;
byte t;

if (true) readi(1);
if (true)
{
    readi(2);
    if (false)
    {
        readi(4);
        print("wow");
        printi(3);
        printi(3 b);
        printi(z);
        printi(t);
    }
}
else readi(5);

if (true)
{
    readi(2);
    if (false)
    {
        readi(4);
        print("wow");
        printi(3);
        printi(3 b);
        printi(z);
        printi(t);
    }
    else
    {
        if(true)
        {
            readi(5);
            print("wow");
            printi(3);
            printi(3 b);
            printi(z);
            printi(t);
        }
        while (true) printi(20);
        while (false)
        {
            readi(5);
            print("wow");
            printi(3);
            printi(3 b);
            printi(z);
            printi(t);
        }
    }
}
else
{
    readi(6);
    print("wow");
    printi(3);
    printi(3 b);
    printi(z);
    printi(t);
}

int x = readi(5);
byte y = (byte)readi(4);
x = (int)readi(1);

x = (byte)readi(1) + readi(2);
x = (byte)(readi(1) + readi(2));
x = readi(1) + (byte)readi(2);
x = (byte)(readi(1) + readi(2));
x = (byte)(readi(1) + (byte)readi(2));
x = (byte)((byte)readi(1) + (byte)readi(2));
x = (byte)(readi(1) + (byte)readi(2));
x = (readi(1) + readi(2));
x = (readi(1) + (byte)readi(2));
x = ((byte)readi(1) + (byte)readi(2));
x = (readi(1) + (byte)readi(2));

y = (byte)(readi(1) + readi(2));
y = (byte)(readi(1) + readi(2));
y = (byte)(readi(1) + (byte)readi(2));
y = (byte)((byte)readi(1) + (byte)readi(2));
y = (byte)(readi(1) + (byte)readi(2));
y = ((byte)readi(1) + (byte)readi(2));

x = (int)readi(1) + readi(2);
x = (int)(readi(1) + readi(2));
x = readi(1) + (int)readi(2);
x = (int)(readi(1) + readi(2));
x = (int)(readi(1) + (int)readi(2));
x = (int)((int)readi(1) + (int)readi(2));
x = (int)(readi(1) + (int)readi(2));
x = (readi(1) + readi(2));
x = (readi(1) + (int)readi(2));
x = ((int)readi(1) + (int)readi(2));
x = (readi(1) + (int)readi(2));

int a = 1 + 1;
a = 1 b + 2;
a = 1 + 2 b;
a = 1 b + 1 b;

byte b1 = 1 b + 1 b;
byte b2 = (byte)(1 + 1);

a = 1 / 1;
a = 1 b / 2;
a = 1 / 2 b;
a = 1 b / 1 b;

a = 1 * 1;
a = 1 b * 2;
a = 1 * 2 b;
a = 1 b * 1 b;

a = 1 - 1;
a = 1 b - 2;
a = 1 - 2 b;
a = 1 b - 1 b;

b1 = 1 b / 1 b;
b2 = (byte)(1 / 1);

b1 = 1 b * 1 b;
b2 = (byte)(1 * 1);

b1 = 1 b - 1 b;
b2 = (byte)(1 - 1);

bool bol = true;
if(bol and true and false or true) printi(3);