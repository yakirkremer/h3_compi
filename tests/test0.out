---end scope---
---end scope---
---end scope---
---end scope---
---end scope---
---end scope---
---end scope---
---end scope---
---end scope---
---end scope---
---end scope---
---end scope---
---end scope---
---end scope---
---end scope---
---end scope---
---end scope---
---end scope---
---end scope---
---end scope---
---end scope---
print (STRING)->VOID 0
printi (INT)->VOID 0
readi (INT)->INT 0
z INT 0
t BYTE 1
x INT 2
y BYTE 3
a INT 4
b1 BYTE 5
b2 BYTE 6
bol BOOL 7
